optax/_src/transform.py

```diff
@@ -869,6 +869,64 @@ def update_fn(updates, state, params=None):
   return base.GradientTransformation(init_fn, update_fn)
 
 
+class ScaleByNewtonState(NamedTuple):
+  """State holding the inverse of the sum of gradient squares to date."""
+
+  hessian_inv: base.Updates
+
+
+def sherman_morrison(a_inv, u, v):
+  den = 1.0 + (u @ a_inv) @ v
+  a_inv -= a_inv @ jnp.outer(u, v) @ a_inv / den
+  return a_inv
+
+
+def scale_by_newton(eps: float = 1e-7) -> base.GradientTransformation:
+  """Rescale updates by multiplying by the inverse of an approximation of the
+  hessian.
+
+  References
+  ----------
+  [^1]: [Hazan, E., Agarwal, A. and Kale, S., 2007. Logarithmic regret
+  algorithms for online convex optimization. Machine Learning, 69(2-3),
+  pp.169-192](https://www.cs.princeton.edu/~ehazan/papers/log-journal.pdf)
+
+  Args:
+    eps: A small floating point value to avoid zero denominator.
+
+  Returns:
+    An (init_fn, update_fn) tuple.
+  """
+
```

Comment:
Note: I haven't reviewed the rest of this file yet as I'm not sure which equations exactly they implement in the paper. Please point me to the algorithm, equation, or section. Thanks a lot!

Reply:
Hi @mkunesch, thanks for your review! 
For your last question, the algorithm implemented is the ONS (ONLINE NEWTON STEP) Fig.2 in the paper.

Reply:
Amazing, thanks a lot - now I understand! I have a few follow-up questions but I'll ask those for specific lines.
(It might be worth adding this to the doc string just because there is additional boilerplate in the paper and it takes a while to see why only the terms in the code remain - thanks a lot!)